File: 2106-find-greatest-common-divisor-of-array/find-greatest-common-divisor-of-array.py
class Solution:
    def findGCD(self, nums: List[int]) -> int:
        def gcd(a, b):
            if a == 0:
                return b
            return gcd(b % a, a)
        a = max(nums)
        b = min(nums)
        return gcd(a, b)

File: 2106-find-greatest-common-divisor-of-array/find-greatest-common-divisor-of-array.java
import java.util.Arrays;

class Solution {
    public int findGCD(int[] nums) {
        int a = Arrays.stream(nums).min().getAsInt();
        int b = Arrays.stream(nums).max().getAsInt();
        return gcd(a, b);
    }

    public int gcd(int a, int b){
        if (a == 0) return b;
        return gcd(b % a, a);
    }
}